### ISSUE TITLE
Training is not stopping

### ISSUE THREAD

**Issue by taborzbislaw** [MIC-DKFZ/nnUNet#389]
Hi,

I am training nnUNet for about two days for 2D data. There are clear signs of over fitting but the training is continued. 
I was trying to stop the training manually, changing the name of model files, then running validation and prediction but it does not work. Is there any way to set a number of training epochs and not to wait until the training is stopped by preset criteria?

![progress](https://user-images.githubusercontent.com/60366744/99089243-533d9400-25cd-11eb-804d-bbcc33609147.png)

Regards
Zbisław

**Comment by FabianIsensee** [comment]
Hi,
yes there is overfitting but this is no problem at all. As long as the red line does not start to increase and as long as the green line does not start to drop everything is fine and you can completely ignore the overfitting part. This is not going to degrade your segmentation accuracy at all.
If you really want to stop early, cancel the training and rename the files to model_final_checkpoint.model and model_final_checkpoint.model.pkl
Best,
Fabian

**Comment by taborzbislaw** [comment]
Dear Fabian,

I have stopped the training and renamed the files as you have advised. However prediction is not running - below is the screen shot with the errors raised, you will see that the model files are properly renamed. 
I am also not able to run validation (before running prediction) after issuing a command e.g.: nnUNet_train -val 2d nnUNetTrainerV2 Task001_Abdomen 2 - see the second print screen

Best regards,
Zbislaw

![Zrzut ekranu z 2020-11-13 17-33-03](https://user-images.githubusercontent.com/60366744/99096316-69038700-25d6-11eb-9378-4acb39bad7a7.png)

![Zrzut ekranu z 2020-11-13 17-37-03](https://user-images.githubusercontent.com/60366744/99096704-e4fdcf00-25d6-11eb-9fe8-5cf9c635d710.png)

**Comment by FabianIsensee** [comment]
you misplaced the -val argument. 

`nnUNet_train 2d nnUNetTrainerV2 Task001_Abdomen 2 -val`

is the correct command

**Comment by taborzbislaw** [comment]
Hi,

my command was:

nnUNet_train -val 2d nnUNetTrainerV2 Task001_Abdomen 2

so I have put -val after nnUNet_train, not at the end of the command. Does it matter? 
And I have renamed model_best.model to model_final_checkpoint.model. Is that correct or should I rename  model_latest.model to model_final_checkpoint.model and accordingly for *.pkl files?
From debug.json it follows that the maximum number of training epochs is 1000. If I change 1000 to say 300 and restart training with 
nnUNet_train -c 2d nnUNetTrainerV2 Task001_Abdomen 2
will the training stop after 300 epochs?

Regards
Z.

**Comment by FabianIsensee** [comment]
Yes it does matter, that's why I pointed it out ;-) nnUNetTrainerV2, Task001_Abdomen and 2 are positional arugments, -val is not
Best,
Fabian

**Comment by FabianIsensee** [comment]
again, -c is also positional so please put it at the end

**Comment by taborzbislaw** [comment]
Ok, thank you. I have changed maximum number of training epochs in debug.json to 300 and started retraining. Tomorrow I will see if the training is stopped.

Best regards,
Zbisław

**Comment by taborzbislaw** [comment]
Dear Fabian,

I have gone through training once more. I have stopped it after 100 epochs and renamed model_latest.model to model_final_checkpoint.model and model_latest.model.pkl to model_final_checkpoint.model.pkl (as it follows from your code that the former are replace by the latter).
I have successfully run validation and selection of the best model running the command:

nnUNet_find_best_configuration -m 2d -t 001 --strict

Then, I was instructed to run the following command for prediction:

nnUNet_predict -i /net/people/plgztabor/nnUNet/TestCases -o /net/people/plgztabor/nnUNet/Results -tr nnUNetTrainerV2 -ctr nnUNetTrainerV2CascadeFullRes -m 2d -p nnUNetPlansv2.1 -t Task001_Abdomen

After running it I have got an error as in the screen shot below:

![Zrzut ekranu z 2020-11-15 16-00-52](https://user-images.githubusercontent.com/60366744/99188498-a09c3b80-275c-11eb-8812-fce0ba1352fc.png)

It appears that nnUNet is saving some additional files which are created only after finishing the training (after 1000 epochs) and if these files are nor present the prediction cannot be launched. Do you have any suggestions how to overcome this difficulty after early stopping? Maybe some template file put into the folders will help?

Best regards,
Zbisław

**Comment by taborzbislaw** [comment]
I have also tried to run prediction from Model_latest using -chk option for nnUNet_predict:

nnUNet_predict -i /net/people/plgztabor/nnUNet/TestCases -o /net/people/plgztabor/nnUNet/Results -m 2d -t Task001_Abdomen -chk model_latest

but it also does not work and the same error is reported.

Z.

**Comment by taborzbislaw** [comment]
Dear Fabian,

prediction is already working - I have finally run it on a machine with a fully configured GPU and then no errors were raised. I have used -chk option of nnUNet_predict so renaming of model files was not necessary after early stopping the training. Thank you for your assistance.

Best regards,
Zbisław
